Expensify/App
# [HOLD Expensify.cash 2196] Refactor login form into a single component

NOTE: waiting on #2196 to get merged first, that's why the diff is so large.

### Details
Before, the `LoginForm` component had narrow and wide counterparts (sub-components), which made the logic confusing.

In lieu of the changes in https://github.com/Expensify/Expensify.cash/pull/2196, we can now refactor this into a single component without these counterparts and clean up the code.

### Fixed Issues
Expensify/Expensify#157816

### Tests
1. Navigate to the sign in page.
2. Verify it renders the same as it does on production.

### QA Steps
1. Navigate to the sign in page.
2. Verify the welcome text everything renders correctly.

### Tested On


### Screenshots

#### Web
<img width="1696" alt="Screen Shot 2021-04-02 at 11 03 58 AM" src="https://user-images.githubusercontent.com/31285285/113375947-271b8780-93a3-11eb-83a5-bc3f1e6836c2.png">

#### Mobile Web
<img width="683" alt="Screen Shot 2021-04-01 at 11 37 28 AM" src="https://user-images.githubusercontent.com/31285285/113240139-b317ab80-92de-11eb-9df2-76e767b066d8.png">

#### Desktop
<img width="1167" alt="Screen Shot 2021-04-02 at 11 00 34 AM" src="https://user-images.githubusercontent.com/31285285/113375957-2da9ff00-93a3-11eb-9a32-1d03c6776dd9.png">

#### iOS
<img width="453" alt="Screen Shot 2021-04-02 at 10 59 53 AM" src="https://user-images.githubusercontent.com/31285285/113375970-38fd2a80-93a3-11eb-882b-64e175927579.png">

#### Android
<img width="391" alt="Screen Shot 2021-04-02 at 11 03 18 AM" src="https://user-images.githubusercontent.com/31285285/113375967-36023a00-93a3-11eb-9c78-ad044e728857.png">
